tests/testcase_generators/testcase_generator.py

```diff
@@ -353,12 +365,27 @@ def generate_test_case(self):
             raise ValueError(
                 "Cannot find inputdata values resulting in non-empty data."
             )
+
         parameters = input_["parameters"]
-        output = self.compute_expected_output(input_data, parameters)
+        try:
+            output = self.compute_expected_output(input_data, parameters)
+        except Exception as err:
+            raise Exception(
+                f"\033[1m\033[91mAn error occurred:\033[0m\033[0m {err}, datasets: \033[1m {input_['inputdata']['datasets']} \033[0m"
```

Comment:
❤️ the visual aid!

Very very minor/ not important: In this case of error propagation, in this exception you should only add the datasets/input information. In the `print` in line 387 you should add the "An error occured: ..." + the visualization because this is not the only error that could arise in the future, so the visual aid wouldn't be there!!! 😨